app/src/main/java/com/example/myapplication/InfoFragment.kt

```diff
@@ -24,6 +31,18 @@ class InfoFragment : Fragment() {
 
     private val args: InfoFragmentArgs by navArgs()
 
+    private val viewModel by viewModels<InfoFragmentViewModel> {
+        object : ViewModelProvider.Factory {
+            override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+                @Suppress("UNCHECKED_CAST")
+                return InfoFragmentViewModel(
+                    rickApi = RickService.provideRetrofit().create(),
```

Comment:
цель провайдера предоставлять зависимости уже конечные
```suggestion
                    rickApi = RickService.provideRickApi(),
```